### ISSUE TITLE
First test run with issues

### ISSUE THREAD

**Issue by nylander** [percyfal/manticore-nf#3]
### Description of the bug

First attempts using the command below generates a number of errors due to different causes.

### Command used and terminal output

```console
nextflow run . -profile test,mamba -qs 4 --sample_sets tests/resources/sample_sets.tsv --window_sizes 1000
```
Generates the following:

```
ERROR: Validation of pipeline parameters failed!

* --window_sizes: expected type: String, found: Integer (1000)

WARN: Found unexpected parameters:
* --enable_conda: true
- Ignore this warning: params.schema_ignore_params = "enable_conda" 
```

The file `nextflow_schema.json` specifies the `window_sizes` as type `string` but changing the `--window_sizes` parameter to `"1000"` did not change the behavior. The schema file also says that the argument should be a comma-separated list. How is that list treated if a single value is provided (I haven't checked)?

Skipping the last `--window_sizes 1000` argument avoids the schema error.

I have not looked more closely on the second warning ("Found unexpected parameters").

# Second issue

Using the same command as above (but skipping the `--window_sizes 1000`) I run in to a mamba-related error with messages similar to this issue: <https://github.com/mamba-org/mamba/issues/1446>.

After adding `disable_lockfile: true` to my `~/.condarc`, the run completed without errors.

I still need to confirm and repeat these errors, but if the "solution" is for users to edit their local settings, this is truly problematic for reproducibility. However, it is part of the challenge developing in nextflow: there are lot of moving parts that may or may not be in sync...

### System information

- Nextflow version 22.10.4 build 5836
- conda version 22.11.1
- Ubuntu 22.04.1 LTS
- openjdk version "18.0.2-ea"

**Comment by percyfal** [comment]
Thanks for the first issue report :+1: My bad - the window_size bug has actually been fixed, but on a dev branch. I'm merging as we write, will close issue when done.

Wrt mamba I didn't have to take any special measures. I'm using conda for testing, but since this is seen [as a last resort](https://nf-co.re/docs/usage/configuration#basic-configuration-profiles) I hope to move testing and running to singularity.

**Comment by percyfal** [comment]
Ok, so I updated Nextflow and now I'm getting the same issues with lockfiles. Running with a single core (-qs 1) seems to do the trick on the first run so some concurrency issue going on here. Once environments have been installed running multiple cores seems to work fine (don't forget to set the environment variable NXF_CONDA_CACHEDIR  to ~/.nextflow/conda or something similar.

**Comment by percyfal** [comment]
See also https://github.com/nextflow-io/nextflow/issues/1819

**Comment by percyfal** [comment]
Fixed via #7
